src/System.Runtime/tests/System/Guid.cs

```diff
@@ -176,4 +177,31 @@ public static void TestToString()
         Assert.Equal(_testGuid.ToString("P"), "(a8a110d5-fc49-43c5-bf46-802db8f843ff)");
         Assert.Equal(_testGuid.ToString("X"), "{0xa8a110d5,0xfc49,0x43c5,{0xbf,0x46,0x80,0x2d,0xb8,0xf8,0x43,0xff}}");
     }
+
+    [Fact]
+    public static void TestRandomness()
+    {
+        const int Iterations = 100;
+        const int GuidSize = 16;
+        byte[] random = new byte[GuidSize * Iterations];
+
+        for (int i = 0; i < Iterations; i++)
+        {
+            // Get a new Guid
+            Guid g = Guid.NewGuid();
+            byte[] bytes = g.ToByteArray();
+
+            // Make sure it's different from all of the previously created ones
+            for (int j = 0; j < i; j++)
+            {
+                Assert.False(bytes.SequenceEqual(new ArraySegment<byte>(random, j * GuidSize, GuidSize)));
+            }
+
+            // Copy it to our randomness array
+            Array.Copy(bytes, 0, random, i * GuidSize, GuidSize);
+        }
+
+        // Verify the randomness of the data in the array
+        RandomDataGenerator.VerifyRandomDistribution(random);
```

Comment:
The 6 fixed bits are, IIRC 0100 (version 4) and 10 (variant 2?), meaning there's already a slight bias toward zeroes.  To put it on even ground with truly random maybe we need to add a bias factor parameter.  In this case, pretend we're starting off having generated two 1 bits.

Or not.  I'm just speaking from a theoretical position :smile:.